src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGeneratorTest.java

```diff
@@ -302,10 +306,42 @@ public void testLastImport() {
         final int lineNumber = 8;
         final int columnNumber = 0;
         final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(rootAst, lineNumber,
-                columnNumber);
+                columnNumber, fileName);
         final List<String> actual = queryGenerator.generate();
         final List<String> expected = Collections.singletonList(
                 "/IMPORT[./DOT[@text='Iterator']]");
         assertEquals("Generated queries do not match expected ones", expected, actual);
     }
+
+    @Test
+    public void testTabWidth() throws Exception {
+        final String filePath = getPath("InputXpathTabWidth.java");
+        final File file = new File(filePath);
+        final DetailAST detailAst = TestUtils.parseFile(file);
+        final int lineNumber = 4;
+        final int columnNumber = 12;
+        final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(detailAst, lineNumber,
+                columnNumber, filePath, 4);
```

Comment:
extract the magic number 4 to the local variable. Think about the person who will support your code.

Reply:
ok

Reply:
done